python/ray/data/aggregate/_aggregate.py

```diff
@@ -41,18 +43,16 @@ class AggregateFn:
             accumulate_block must be provided.
         finalize: This is called once to compute the final aggregation
             result from the fully merged accumulator.
-        name: The name of the aggregation. This will be used as the output
-            column name in the case of Arrow dataset.
     """
 
     def __init__(
         self,
         init: Callable[[KeyType], AggType],
         merge: Callable[[AggType, AggType], AggType],
+        name: str,
         accumulate_row: Callable[[AggType, T], AggType] = None,
         accumulate_block: Callable[[AggType, Block], AggType] = None,
         finalize: Callable[[AggType], U] = lambda a: a,
-        name: Optional[str] = None,
```

Comment:
Did `AggregateFn` always error if `name` isn't specified? If there are cases where `AggregateFn` worked without a name, this'd be a breaking change.

Reply:
all the existing constructor usages in ray specify a name. i think the error occurs when we call `aggregate` on a `GroupedDataset` without providing a name. 

if there are external users using this without a name, it could potentially break, but i think throwing a `TypeError` specific for this input should suffice.